### DISCUSSION TITLE
found bug -- how to create a branch and pull request?

### DISCUSSION THREAD

**Discussion by bayesfactor** [openai/whisper#1467] [Q&A]
There is a bug with the use of `Optional` in:  https://github.com/openai/whisper/blob/248b6cb124225dd263bb9bd32d060b6517e067f8/whisper/utils.py#L104

The fix is easy, how can I create a branch and a PR? I fixed it in my own fork but I don't have permissions to open an issue or create a branch in this repo. Here is the patch:
https://github.com/bayesfactor/whisper/pull/1/commits/34a6edb783fceec1eda496fd5a50f2400c87c257

**Comment by phineas-pta** [comment]
https://letmegooglethat.com/?q=how+to+create+github+pr

also better use `options.get("......")` instead of `options["...."] if .....`

**Reply by glangford** [reply]
https://stackoverflow.com/questions/6130768/return-a-default-value-if-a-dictionary-key-is-not-available

**Reply by bayesfactor** [reply]
`options.get("...")` is the right approach, good suggestion. I know how to create a branch & PR in other repos; I don't have permission to open an issue or create a branch in this repo.

**Reply by phineas-pta** [reply]
of course u cannot create a branch in a repo not owned by u, but no one can stop u from creating PR, did u really read any tutorials ? like the github docs below ?

**Comment by NielsMayer** [comment]
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork
https://opensource.com/article/19/7/create-pull-request-github

**Reply by bayesfactor** [reply]
The [opensource link](https://opensource.com/article/19/7/create-pull-request-github1) is a dead link. I should have clarified -- I know how to create a fork and pull request, but I cannot in this repo -- I suspect I don't have permission?

**Reply by NielsMayer** [reply]
an extra "1" got appended to the URL by accident.... correct URL is 
https://opensource.com/article/19/7/create-pull-request-github

> Find a project you want to contribute to
> Fork it
> Clone it to your local system
> Make a new branch
> Make your changes
> Push it back to your repo
> Click the Compare & pull request button
> Click Create pull request to open a new pull request

**Comment by bayesfactor** [comment]
ok, here is a PR: https://github.com/openai/whisper/pull/1476

**Reply by phineas-pta** [reply]
`options.get("highlight_words", False)`
